bundles/org.openhab.binding.systeminfo/src/main/java/org/openhab/binding/systeminfo/internal/SystemInfoHandlerFactory.java

```diff
@@ -41,24 +45,30 @@ public class SystemInfoHandlerFactory extends BaseThingHandlerFactory {
     private @NonNullByDefault({}) SystemInfoInterface systeminfo;
     private @NonNullByDefault({}) SystemInfoThingTypeProvider thingTypeProvider;
 
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS;
+    static {
+        Set<ThingTypeUID> buffer = new HashSet<>();
+        buffer.add(THING_TYPE_COMPUTER);
+        buffer.add(THING_TYPE_COMPUTER_IMPL);
+        SUPPORTED_THING_TYPES_UIDS = Collections.unmodifiableSet(buffer);
+    }
+
```

Comment:
Could be replaced by `private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(THING_TYPE_COMPUTER,THING_TYPE_COMPUTER_IMPL);`

Reply:
On purpose this way: https://github.com/openhab/openhab-addons/commit/af37fe704fcfc2cc3040d049e63868f4c491d0f6 I am not sure, if there is an intereset from the community, to refactor the binding model. This will lead in non backward compatible changes. The linked commit extracts hard drive interface only, but it can be done very easy for other device types.

Reply:
I think @clinique only suggested to improve the initialization of the static `Set`. This is similar to the change in #16763.

Reply:
Changed: i simply forgot varargs constructor.